### PR TITLE
Feature/currency strict constructor

### PR DESCRIPTION
**Pull request by jonbern**
### Description
Enables strict type checking for `id` property in Currency constructor.

Enables test cases unintentionally disabled/not included for the `Currency.from()` constructor methods tests. Updated test helpers to cover all edge cases defined by the now included test cases.

### Issues
Issues: 

### Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] The [Contribution Guidelines][1] has been followed
- [x] Tests for the changes have been added and all test pass (`npm test`)
- [x] JSDoc comments have been reviewed and added / updated as needed
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures
- [x] Package version number was updated according to semantic rules
- [x] Any changes to upstream dependencies have already been published

[1]: https://github.com/hubiinetwork/nahmii-sdk#contributing

### CODE REVIEW

File: lib/currency.spec.js [hunk 1]
```diff
@@ -111,12 +137,13 @@ describe('Currency', () => {
         }
     });
 
-    context('Factory return null on illegal input', () => {
-        for (const [ct, id] of badConstructorArgs) {
-            it (`with "${ct}" ${typeStr(ct)}, "${id}" ${typeStr(id)}`, () => {
-                expect(Currency.from({ct, id})).to.equal(null);
+    context('Factory returns null on illegal input', () => {
```
Comment (morfj) [inline lib/currency.spec.js]: This test sounds inverted (context should define the scenario, inputs, etc, *not* the expected result). Not a showstopper but something to keep in mind for the next interation.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nahmii-sdk",
-  "version": "3.1.1",
+  "version": "3.1.2",
```
Comment (morfj) [inline package.json]: I think this is OK since there are hardly any users at present. The constructor change *might* be considered a breaking change for some though!?!?

Comment (jonbern) [reply package.json]: I get your point, and I thought about it myself. The way I see it is that we are now fixing a bug in the original implementation, and thus a patch release is appropriate.

File: lib/currency.spec.js (hunk 1 above)
Comment (jonbern) [reply lib/currency.spec.js]: I think that makes a lot of sense!
